src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeLift.java

```diff
@@ -418,73 +122,332 @@ public static CoreOps.FuncOp liftToBytecodeDialect(byte[] classdata, String meth
-        });
 
-        return f;
-    }
+            // Process blocks
+            for (BytecodeBasicBlock bcb : bytecodeBody.blocks) {
+                Block.Builder b = c.blockMap.get(bcb);
+                // If a non-entry block has parameters then they correspond to stack arguments
+                // Push back block parameters on the stack, in reverse order to preserve order on the stack
+                Deque<Value> stack = new ArrayDeque<>();
+                if (!b.parameters().isEmpty() && !b.isEntryBlock()) {
+                    for (int i = b.parameters().size() - 1; i >= 0; i--) {
+                        stack.push(b.parameters().get(i));
+                    }
+                }
+                int ni = bcb.instructions.size();
+                for (int i = 0; i < ni; i++) {
+                    switch (bcb.instructions.get(i)) {
+                        case LabelTarget labelTarget -> {
+        // @@@                    for (ExceptionCatch tryCatchBlock : bcr.codeModel.exceptionHandlers()) {
+        //                        if (labelTarget.label() == tryCatchBlock.tryStart()) {
+        //                            BytecodeBasicBlock handler = bcr.blockMap.get(tryCatchBlock.handler());
+        //                            b.op(BytecodeInstructionOps.
+        //                                    exceptionTableStart(c.blockMap.get(handler).successor()));
+        //                        } else if (labelTarget.label() == tryCatchBlock.tryEnd()) {
+        //                            b.op(BytecodeInstructionOps.exceptionTableEnd());
+        //                        }
+        //                    }
+                        }
+                        case LineNumber ln -> {
+                            // @@@ Add special line number instructions
+                        }
+                        case LocalVariable lv -> {
+                            // @@@
+                        }
+                        case LocalVariableType lvt -> {
+                            // @@@
+                        }
+                        case LoadInstruction inst -> {
+                            stack.push(b.op(CoreOps.varLoad(locals.get(inst.slot()))));
+                        }
+                        case StoreInstruction inst -> {
+                            Value operand = stack.pop();
+                            Op.Result local = locals.get(inst.slot());
+                            if (local == null) {
+                                local = b.op(CoreOps.var(Integer.toString(lvm), operand));
+                                locals.put(lvm++, local);
+                            } else {
+                                TypeDesc varType = ((CoreOps.VarOp) local.op()).varType();
+                                if (!operand.type().equals(varType)) {
+                                    local = b.op(CoreOps.var(Integer.toString(lvm), operand));
+                                    locals.put(lvm++, local);
```

Comment:
That looks incorrect (probably always was). The slot is reused with a different type so we need to update the existing entry in the map. This likely always connects to how to manage the map with conditional branching. Suggest adding an `@@@` comment for now.

Reply:
Addd, will revisit it later, thanks.